File: java-kotlin-katas/src/main/katas/lupe/codewars/katas/java/piglatin/PigLatin.java
package lupe.codewars.katas.java.piglatin;

import java.util.Arrays;
import java.util.stream.Collectors;

public class PigLatin {
    public String translate(String str) {
        return str.isEmpty() ? "" : Arrays.stream(str.split(" "))
                .filter(el -> el.length() >= 1)
                .map(el -> el.charAt(0) == '!'
                        || el.charAt(0) == '?'
                        || el.charAt(0) == '.' ? el
                        : el.substring(1) + el.charAt(0) + "ay")
                .collect(Collectors.joining(" "));
    }
}


// BEST PRACTICE from Codewars //

/*

public static String pigIt(String str) {
    return str.replaceAll("(\\w)(\\w*)", "$2$1ay");
}

*/

File: java-kotlin-katas/src/main/katas/lupe/codewars/katas/kotlin/piglatin/PigLatin.kt
package lupe.codewars.katas.kotlin.piglatin

import java.util.stream.Collectors

class PigLatin {
    fun translate(str: String): String {
        return str.split(" ").stream()
            .filter { it.isNotEmpty() }
            .map { if (it.startsWith('!')
                || it.startsWith('?')
                || it.startsWith('.')) it
                else it.substring(1) + it.first() + "ay" }
            .collect(Collectors.joining(""))
    }
}